src/moin/utils/send_file.py

```diff
@@ -127,24 +132,21 @@ def send_file(
     # We must compute size the smart way rather than letting
     # werkzeug turn our iterable into an in-memory sequence
     # See `_ensure_sequence` in werkzeug/wrappers.py
+    fsize = None
     if filename:
         fsize = os.path.getsize(filename)
+
     elif file and hasattr(file, "seek") and hasattr(file, "tell"):
-        fsize = None
         # be extra careful as some file-like objects (like zip members) have a seek
         # and tell methods, but they just raise some exception (e.g. UnsupportedOperation)
         # instead of really doing what they are supposed to do (or just be missing).
         try:
             file.seek(0, 2)  # seek to EOF
-            try:
-                fsize = file.tell()  # tell position
-            except Exception:
-                pass
+            fsize = file.tell()  # tell position
             file.seek(0, 0)  # seek to start of file
```

Comment:
if it succeeds with seek, fails with tell, it will not seek back to start of file.

but, it will log a warning AND CONTINUE PROCESSING (and malfunction, because it is at EOF position).

Reply:
Added a final seek.